lib/lightning/runs/handlers.ex

```diff
@@ -37,7 +37,8 @@ defmodule Lightning.Runs.Handlers do
     def call(run, params) do
       with {:ok, attrs} <- new(run, params) |> apply_action(:validate),
            {:ok, step} <- insert(attrs) do
-        run = Runs.get(attrs.run_id, include: [:workflow])
+        run = Runs.get(attrs.run_id, include: [:workflow, :snapshot])
+        step = Repo.preload(step, :snapshot)
```

Comment:
@elias-ba we need to come back to this stuff and push the responsibiliy of preloading down to the subscribers of the event.